src/main/java/Duke.java

```diff
@@ -1,10 +1,272 @@
+            printList();
+            startTask();
+        }
+
+        // Switch to Echo mode if user types in "change"
+        if (textLowerC.equals("change")) {
+            switchMode(ECHO_MODE);
+        }
+    }
+
+    /**
+     * Checks different scenarios for input commands of user in Task mode and
+     * does the necessary action.
+     *
+     * @param toAdd String input by user.
+     * @param textLowerC String input by user converted to lower cases for checking.
+     */
+    private static void defaultTaskModeBehavior(String toAdd, String textLowerC) {
+        // Mark task as complete with an X.
+        if (textLowerC.startsWith("completed ") | textLowerC.startsWith("done ")) {
+            int taskNo = Integer.parseInt(toAdd.replaceAll("[^0-9]", "")) - 1;
+            storedTasks[taskNo].setDone();
+            // Print response when task marked done.
+            System.out.println("\t" + HOR_LINE);
+            System.out.printf("\tThat's great! %s has been checked as completed!\n", storedTasks[taskNo].description);
+            System.out.println("\t" + HOR_LINE + System.lineSeparator());
+        }
+        // Remove task from list.
+        else if (textLowerC.startsWith("clear ") | textLowerC.startsWith("remove ")) {
+            int taskNo = Integer.parseInt(toAdd.replaceAll("[^0-9]", "")) - 1;
+            System.out.println("\t" + HOR_LINE);
+            System.out.printf("\t%s removed from list!\n", storedTasks[taskNo].description);
+            System.out.println("\t" + HOR_LINE + System.lineSeparator());
+            System.arraycopy(storedTasks,taskNo + 1, storedTasks, taskNo, additions - taskNo);
+            additions--;
+            printList();
+        }
+        // Create a new task if it does not exist in list
+        else {
+            createTask(toAdd);
+        }
+    }
+
+    /**
+     * Double confirms if user wants to exit.
+     * If no, gets user back to Task/Echo mode user was in.
+     * If yes, ends program with farewell message.
+     */
+    public static void exit() {
+        System.out.print("\tDo you really want to exit chatbot (type y or n)? ");
+        Scanner input = new Scanner(System.in);
+        String exit_pref = input.nextLine();
+
+        // User wishes to leave program.
+        if (exit_pref.equals("y")) {
+            System.out.println("\t" + HOR_LINE);
+            System.out.println("\tBye. Hope to see you again soon!");
+            System.out.println("\t" + HOR_LINE + System.lineSeparator());
+        }
+        else {
+            System.out.println("\n\tOk that's great! Continue keying in commands. :)");
+            System.out.println("\t" + HOR_LINE + System.lineSeparator());
+            // Return back to previous mode since user is not exiting.
+            if (mode == ECHO_MODE) {
+                startEcho();
+            }
+            else if (mode == TASK_MODE) {
+                startTask();
+            }
+            exit();
```

Comment:
Same comment as before - perhaps remove the newline before `else`?

Reply:
Thanks for your comment.